homeassistant/components/portainer/binary_sensor.py

```diff
@@ -0,0 +1,170 @@
+                    coordinator=portainer,
+                    entry=entry,
+                    entity_description=entity_description,
+                    device_info=endpoint,
+                )
+                for entity_description in ENDPOINT_SENSORS
+            ]
+        )
+
+        for container in endpoint.containers.values():
+            entities.extend(
+                [
+                    PortainerContainerSensor(
+                        coordinator=portainer,
+                        entry=entry,
+                        entity_description=entity_description,
+                        device_info=container,
+                        via_device=endpoint,
+                    )
+                    for entity_description in CONTAINER_SENSORS
+                ]
+            )
+
+    async_add_entities(entities, True)
+
+
+class PortainerEndpointSensor(PortainerEndpointEntity, BinarySensorEntity):
+    """Representation of a Portainer endpoint binary sensor entity."""
+
+    entity_description: PortainerBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PortainerCoordinator,
+        entry: PortainerConfigEntry,
+        entity_description: PortainerBinarySensorEntityDescription,
+        device_info: PortainerCoordinatorData,
+    ) -> None:
+        """Initialize Portainer endpoint binary sensor entity."""
+        self.entity_description = entity_description
+        super().__init__(device_info, entry, coordinator)
+
+        self._attr_unique_id = f"{entity_description.key} {device_info.id}"
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        try:
+            assert self.device_id in self.coordinator.endpoints
+            self._device_info = self.coordinator.endpoints[self.device_id]
+        except KeyError:
+            return
+
+        self._attr_is_on = self.entity_description.state_fn(self._device_info)
+        if self.entity_description.attributes_fn is not None:
+            self._attr_extra_state_attributes = self.entity_description.attributes_fn(
+                self._device_info
+            )
+        super()._handle_coordinator_update()
+
+
+class PortainerContainerSensor(PortainerContainerEntity, BinarySensorEntity):
+    """Representation of a Portainer container sensor."""
+
+    entity_description: PortainerBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PortainerCoordinator,
+        entry: PortainerConfigEntry,
+        entity_description: PortainerBinarySensorEntityDescription,
+        device_info: DockerContainer,
+        via_device: PortainerCoordinatorData,
+    ) -> None:
+        """Initialize the Portainer container sensor."""
+        self.entity_description = entity_description
+        super().__init__(device_info, entry, coordinator, via_device)
+
+        self._attr_unique_id = f"{entity_description.key} {device_info.id}"
```

Comment:
This could also go into the base entity class
```suggestion
        self._attr_unique_id = f"{self.coordinator.config_entry.entry_id}_{entity_description.key}_{device_info.id}"
```